drivers/sensor/fxos8700/Kconfig

```diff
@@ -209,6 +209,42 @@ config FXOS8700_MOTION_INT1
 	help
 	  Say Y to route motion interrupt to INT1 pin. Say N to route to INT2 pin.
 
+menuconfig FXOS8700_MAG_VECM
+	bool "Magnetic vector-magnitude detection"
+	help
+	  Enable magnetic vector-magnitude detection
+
+if FXOS8700_MAG_VECM
+
+config FXOS8700_MAG_VECM_INT1
+	bool "Magnetic vector-magnitude interrupt to INT1 pin"
+	help
+	  Say Y to route magnetic vector-magnitude interrupt to INT1 pin.
+	  Say N to route to INT2 pin.
+
+config FXOS8700_MAG_VECM_CFG
+	hex "Magnetic vector-magnitude configuration register"
+	range 0 0x7f
+	default 0x4e
+
+config FXOS8700_MAG_VECM_THS_MSB
+	hex "Magnetic vector-magnitude threshold configuration register"
+	range 0 0x7f
+	default 0x00
+	help
+	  Seven most significant bits of 15-bit unsigned magnetic
+	  vector-magnitude threshold. Resolution is 0.1 Î¼T/LSB.
+
+config FXOS8700_MAG_VECM_THS_LSB
+	hex "Magnetic vector-magnitude threshold configuration register"
+	range 0 0xff
+	default 0x5a
+	help
+	  Eight least significant bits of 15-bit unsigned magnetic
+	  vector-magnitude threshold. Resolution is 0.1 Î¼T/LSB.
```

Comment:
I think that initial hardware configuration should use device tree bindings. Using device tree bindings allows to have per-device settings. Maybe this is not needed for this type of sensor, but who knows on specific applications.

Reply:
Should we then wait for PR #27418 to be merged and realign with that or what do you suggest?

Reply:
No, I don't think that PR will be ready for a bit. I propose we go ahead with this in Kconfig for now, and I can roll in the vector magnitude bits into the larger Kconfig-to-devicetree conversion. @gmarull are you ok with that?

Reply:
I'm ok with that, no problem

Reply:
Thanks. Mind giving your +1 on this PR?